### PR TITLE
Gallery/metadata component

### PR DESCRIPTION
**Pull request by TalMalka123**
- Metadata key-value component implementation- both editable mode and uneditable mode +story. 
   - I added support for a logic that if you click outside of the editable field, or press enter, it saves the changes locally.
   - Also, it is possible to limit the height of the metadata key-value section, and then when creating a new field, the component will scroll down automatically to the newly created field. 
   - In addition, already added support for viewing auto-generated data, which won't be editable).
   - Pressing on the pencil icon, activates the edit mode (if enabled).
   - I added also the option to delete, although it was not in the design, in case we want it. For newly created fields, the delete button is always enabled
   - Validations are not supported yet, and I'm still not sure how should I handle that case of multiple:true.
- I improved and aligned the dropdownV2 component- both CSS-wise and functional-wise. (close when clicking outside etc.)

**Here are some screenshots:**

Edit mode not enabled:
image

Edit mode enabled:
image
image

Edit mode+ Deletion enabled:
image

### CODE REVIEW

File: src/stories/elements/MetadataKeyValueList/MeatadataKeyValueList.stories.tsx
Comment (simonlsk) [inline src/stories/elements/MetadataKeyValueList/MeatadataKeyValueList.stories.tsx:25]: UX suggestions:
1. Make the whole "Add new" row clickable
2. When editing text "Escape" should also "undo" 
3. When adding new item, should have auto focus on the label name
4. When fields don't already exist (new items), the fields should be editable, until I put a value in them.
5. When a row is not filled, clicking "add new" should highlight it.

Comment (simonlsk) [inline src/stories/elements/MetadataKeyValueList/MeatadataKeyValueList.stories.tsx:25]: Same comment as in the metadata tags: I think this belongs inside the data-engine directory, WDYT?

File: src/components/elements/metadataKeyValue/MetadataKeyValueList.tsx
Comment (simonlsk) [inline src/components/elements/metadataKeyValue/MetadataKeyValueList.tsx:47]: This should trigger an external function, for which deletion from the component should succeed only if the external function returned a success value.
For example, 
```
if (onDeleteMetadata(field) { 
  <actually delete> 
} else { 
  alert("failed to delete") 
}
```
Or catch exception or whatever makes sense, we need to decide on a convention for how we do backend validations like that.

File: src/components/elements/metadataKeyValue/MetadataKeyValuePair.tsx
Comment (simonlsk) [inline src/components/elements/metadataKeyValue/MetadataKeyValuePair.tsx:91]: I think this should be more generic. The parent should be able to send a "subtext" or "description". It can be simple text or a component. Then "Auto-generated" is passed by the parent as a prop.

File: src/components/elements/metadataKeyValue/MetadataKeyValuePair.tsx
Comment (simonlsk) [inline src/components/elements/metadataKeyValue/MetadataKeyValuePair.tsx:133]: I think the second one should be named `isRemovable` to be consistent with `isEditable` and also to make it generic. In addition `isAutoGenerated` shouldn't be a thing here, the values of `isEditable` and `isRemovable` should both be false for an item that is auto-generated. That predicating whether the field is editable is decided by the caller component.

File: src/components/elements/metadataKeyValue/CustomTextField.tsx
Comment (simonlsk) [inline src/components/elements/metadataKeyValue/CustomTextField.tsx:8]: I think this component should have it's own story.

**Review by simonlsk — CHANGES_REQUESTED** [review]
This looks really cool, wow!!! 
I wrote my UX comments at the bottom of the story file. In general I think you need to always ask the question "should that component be aware of X". A good example is should an editable text field be aware that it's editing a value of an auto-generated metadata? Probably not. 
The more the component is "pure" the more flexible the design system is, and easier to grow and maintain.

**Comment by TalMalka123** [comment]
**Here is the list of fixes I added:**
- [x] Add a separate story for the CustomTextField component.
- [x] Try-catch on delete and error alert.
- [x] When adding a new field, the key should be auto-focused.
- [x] When editing text, "Escape" should also "undo”, like the X button.
- [x] When clicking on add new button, if there are empty fields, highlight them and don’t create new field.
- [x] For newly created fields, I shouldn’t have the ”pencil” on hover, but I should be able to simply click on it to edit. This behavior is relevant as long as it is empty.
- [x] Make the whole "Add new" row clickable.
- [x] Remove the option to clear value type selection. 
- [x] Move the component to the data-engine folder.
- [x] Don’t add the isAutoGenerated prop, but include it in the isRemovable and is Editable logic.
- [x] Chang variables’ names.

**Things that should be added externally (In nextJs project) to this component are:** 
- Save button and its action.
- Errors messages for miscorrelation between field type and value. (should block the save action from happening).
- Error alerts for failed save and failed deletion.
- Clear definition on how to handle "multiple" field set to true, and multiple values to a single key (if there is such a thing)

**Review by simonlsk — APPROVED** [review]
Overall LGTM. 
The behavior in editing mode is still not smooth and has some issues, but lets put that on hold for now.
In view mode the only problem now is the tooltip. 
* I think it doesn't really match the design language. 
* I don't like the arrow.
* It should show up in delay. 
* It shouldn't hide the next value below.
* I don't see any good reason to have the whole custom handling of `open` and controlling the `onMouseLeave` and `onMouseEnter` behavior. This is a textbook use case for tooltip. 

My intuition is just drop it for now, and ship without it, and if you do the PR is approved.
If you want to tooltip to be part of it, then make sure you let Anna, Dean / Guy see it and play with it before you merge.

**Comment by TalMalka123** [comment]
There should be a following pr to add a tooltip that fits the design, and fix the issues related to edit mode

**Pull request merged**
